Creating and starting the container...
mmcl_rvcl
Removing existing container
Container removed
Creating new container
[INFO] Extracting squashfs filesystem...
Parallel unsquashfs: Using 20 processors
132418 inodes (238794 blocks) to write


created 130336 files
created 17421 directories
created 1971 symlinks
created 0 devices
created 0 fifos
created 0 sockets
Container created
Starting container
                                                                                                                                                
================
== TensorFlow ==
================

NVIDIA Release 20.12-tf1 (build 18410160)
TensorFlow Version 1.15.4

Container image Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
Copyright 2017-2020 The TensorFlow Authors.  All rights reserved.

NVIDIA Deep Learning Profiler (dlprof) Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.

Various files include modifications (c) NVIDIA CORPORATION.  All rights reserved.
NVIDIA modifications are covered by the license terms that apply to the underlying project or file.
WARNING: Detected NVIDIA NVIDIA H100 GPU, which is not yet supported in this version of the container
ERROR: No supported GPU(s) detected to run this container

NOTE: MOFED driver was detected, but nv_peer_mem driver was not detected.
      Multi-node communication performance may be reduced.

Setting up the environment...
Collecting pip
  Using cached pip-25.0.1-py3-none-any.whl (1.8 MB)
Installing collected packages: pip
  Attempting uninstall: pip
    Found existing installation: pip 20.2.4
    Uninstalling pip-20.2.4:
      Successfully uninstalled pip-20.2.4
Successfully installed pip-25.0.1
Found existing installation: onnx 1.8.0
Uninstalling onnx-1.8.0:
  Successfully uninstalled onnx-1.8.0
Collecting onnx==1.9.0
  Using cached onnx-1.9.0-cp38-cp38-manylinux2010_x86_64.whl.metadata (10 kB)
Requirement already satisfied: protobuf in /usr/local/lib/python3.8/dist-packages (from onnx==1.9.0) (3.14.0)
Requirement already satisfied: numpy>=1.16.6 in /usr/local/lib/python3.8/dist-packages (from onnx==1.9.0) (1.17.3)
Requirement already satisfied: six in /usr/local/lib/python3.8/dist-packages (from onnx==1.9.0) (1.13.0)
Requirement already satisfied: typing-extensions>=3.6.2.1 in /usr/local/lib/python3.8/dist-packages (from onnx==1.9.0) (3.7.4.3)
Using cached onnx-1.9.0-cp38-cp38-manylinux2010_x86_64.whl (12.2 MB)
WARNING: Error parsing dependencies of pyzmq: Invalid version: 'cpython'
Installing collected packages: onnx
Successfully installed onnx-1.9.0
Installing PyTorch 1.13.1 and compatible libraries...
Looking in indexes: https://download.pytorch.org/whl/cu117
Collecting torch==1.13.1
  Using cached https://download.pytorch.org/whl/cu117/torch-1.13.1%2Bcu117-cp38-cp38-linux_x86_64.whl (1801.8 MB)
Collecting torchvision==0.14.1
  Using cached https://download.pytorch.org/whl/cu117/torchvision-0.14.1%2Bcu117-cp38-cp38-linux_x86_64.whl (24.3 MB)
Collecting torchaudio==0.13.1
  Using cached https://download.pytorch.org/whl/cu117/torchaudio-0.13.1%2Bcu117-cp38-cp38-linux_x86_64.whl (4.2 MB)
Requirement already satisfied: typing-extensions in /usr/local/lib/python3.8/dist-packages (from torch==1.13.1) (3.7.4.3)
Requirement already satisfied: numpy in /usr/local/lib/python3.8/dist-packages (from torchvision==0.14.1) (1.17.3)
Requirement already satisfied: requests in /usr/local/lib/python3.8/dist-packages (from torchvision==0.14.1) (2.25.0)
Requirement already satisfied: pillow!=8.3.*,>=5.3.0 in /usr/local/lib/python3.8/dist-packages (from torchvision==0.14.1) (6.2.1)
Requirement already satisfied: chardet<4,>=3.0.2 in /usr/local/lib/python3.8/dist-packages (from requests->torchvision==0.14.1) (3.0.4)
Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.8/dist-packages (from requests->torchvision==0.14.1) (2.10)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /usr/local/lib/python3.8/dist-packages (from requests->torchvision==0.14.1) (1.26.2)
Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.8/dist-packages (from requests->torchvision==0.14.1) (2020.11.8)
WARNING: Error parsing dependencies of pyzmq: Invalid version: 'cpython'
Installing collected packages: torch, torchvision, torchaudio
Successfully installed torch-1.13.1+cu117 torchaudio-0.13.1+cu117 torchvision-0.14.1+cu117
Collecting numpy==1.18.5 (from -r requirements.txt (line 1))
  Using cached numpy-1.18.5-cp38-cp38-manylinux1_x86_64.whl.metadata (2.1 kB)
Requirement already satisfied: tqdm in /usr/local/lib/python3.8/dist-packages (from -r requirements.txt (line 2)) (4.54.0)
Collecting scikit-learn==1.0.2 (from -r requirements.txt (line 3))
  Using cached scikit_learn-1.0.2-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (10 kB)
Collecting diffdist==0.1 (from -r requirements.txt (line 4))
  Using cached diffdist-0.1-py3-none-any.whl
Collecting matplotlib==3.4.2 (from -r requirements.txt (line 5))
  Using cached matplotlib-3.4.2-cp38-cp38-manylinux1_x86_64.whl.metadata (5.7 kB)
Collecting appdirs (from -r requirements.txt (line 6))
  Using cached appdirs-1.4.4-py2.py3-none-any.whl.metadata (9.0 kB)
Collecting oslo_concurrency (from -r requirements.txt (line 7))
  Using cached oslo.concurrency-6.1.0-py3-none-any.whl.metadata (2.1 kB)
Collecting pandas==1.2.4 (from -r requirements.txt (line 8))
  Using cached pandas-1.2.4-cp38-cp38-manylinux1_x86_64.whl.metadata (4.7 kB)
Collecting scipy==1.7.3 (from -r requirements.txt (line 9))
  Using cached scipy-1.7.3-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (2.2 kB)
Collecting statsmodels==0.13.5 (from -r requirements.txt (line 10))
  Using cached statsmodels-0.13.5-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (9.5 kB)
Requirement already satisfied: joblib>=0.11 in /usr/local/lib/python3.8/dist-packages (from scikit-learn==1.0.2->-r requirements.txt (line 3)) (0.17.0)
Requirement already satisfied: threadpoolctl>=2.0.0 in /usr/local/lib/python3.8/dist-packages (from scikit-learn==1.0.2->-r requirements.txt (line 3)) (2.1.0)
Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.8/dist-packages (from matplotlib==3.4.2->-r requirements.txt (line 5)) (0.10.0)
Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.8/dist-packages (from matplotlib==3.4.2->-r requirements.txt (line 5)) (1.3.1)
Requirement already satisfied: pillow>=6.2.0 in /usr/local/lib/python3.8/dist-packages (from matplotlib==3.4.2->-r requirements.txt (line 5)) (6.2.1)
Requirement already satisfied: pyparsing>=2.2.1 in /usr/local/lib/python3.8/dist-packages (from matplotlib==3.4.2->-r requirements.txt (line 5)) (2.4.7)
Requirement already satisfied: python-dateutil>=2.7 in /usr/local/lib/python3.8/dist-packages (from matplotlib==3.4.2->-r requirements.txt (line 5)) (2.8.1)
Requirement already satisfied: pytz>=2017.3 in /usr/local/lib/python3.8/dist-packages (from pandas==1.2.4->-r requirements.txt (line 8)) (2020.4)
Collecting patsy>=0.5.2 (from statsmodels==0.13.5->-r requirements.txt (line 10))
  Using cached patsy-1.0.1-py2.py3-none-any.whl.metadata (3.3 kB)
Collecting packaging>=21.3 (from statsmodels==0.13.5->-r requirements.txt (line 10))
  Using cached packaging-24.2-py3-none-any.whl.metadata (3.2 kB)
Collecting fasteners>=0.7.0 (from oslo_concurrency->-r requirements.txt (line 7))
  Using cached fasteners-0.19-py3-none-any.whl.metadata (4.9 kB)
Collecting oslo.config>=5.2.0 (from oslo_concurrency->-r requirements.txt (line 7))
  Using cached oslo.config-9.6.0-py3-none-any.whl.metadata (3.0 kB)
Collecting oslo.i18n>=3.15.3 (from oslo_concurrency->-r requirements.txt (line 7))
  Using cached oslo.i18n-6.4.0-py3-none-any.whl.metadata (2.1 kB)
Collecting oslo.utils>=3.33.0 (from oslo_concurrency->-r requirements.txt (line 7))
  Using cached oslo.utils-7.3.0-py3-none-any.whl.metadata (2.3 kB)
Collecting pbr>=2.0.0 (from oslo_concurrency->-r requirements.txt (line 7))
  Using cached pbr-6.1.1-py2.py3-none-any.whl.metadata (3.4 kB)
Requirement already satisfied: six in /usr/local/lib/python3.8/dist-packages (from cycler>=0.10->matplotlib==3.4.2->-r requirements.txt (line 5)) (1.13.0)
Requirement already satisfied: PyYAML>=5.1 in /usr/local/lib/python3.8/dist-packages (from oslo.config>=5.2.0->oslo_concurrency->-r requirements.txt (line 7)) (5.3.1)
Collecting debtcollector>=1.2.0 (from oslo.config>=5.2.0->oslo_concurrency->-r requirements.txt (line 7))
  Using cached debtcollector-3.0.0-py3-none-any.whl.metadata (2.3 kB)
Collecting netaddr>=0.7.18 (from oslo.config>=5.2.0->oslo_concurrency->-r requirements.txt (line 7))
  Using cached netaddr-1.3.0-py3-none-any.whl.metadata (5.0 kB)
Requirement already satisfied: requests>=2.18.0 in /usr/local/lib/python3.8/dist-packages (from oslo.config>=5.2.0->oslo_concurrency->-r requirements.txt (line 7)) (2.25.0)
Collecting rfc3986>=1.2.0 (from oslo.config>=5.2.0->oslo_concurrency->-r requirements.txt (line 7))
  Using cached rfc3986-2.0.0-py2.py3-none-any.whl.metadata (6.6 kB)
Collecting stevedore>=1.20.0 (from oslo.config>=5.2.0->oslo_concurrency->-r requirements.txt (line 7))
  Using cached stevedore-5.3.0-py3-none-any.whl.metadata (2.3 kB)
Collecting iso8601>=0.1.11 (from oslo.utils>=3.33.0->oslo_concurrency->-r requirements.txt (line 7))
  Using cached iso8601-2.1.0-py3-none-any.whl.metadata (3.7 kB)
Collecting netifaces>=0.10.4 (from oslo.utils>=3.33.0->oslo_concurrency->-r requirements.txt (line 7))
  Using cached netifaces-0.11.0-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.whl.metadata (9.0 kB)
Requirement already satisfied: setuptools in /usr/local/lib/python3.8/dist-packages (from pbr>=2.0.0->oslo_concurrency->-r requirements.txt (line 7)) (50.3.2)
Requirement already satisfied: wrapt>=1.7.0 in /usr/local/lib/python3.8/dist-packages (from debtcollector>=1.2.0->oslo.config>=5.2.0->oslo_concurrency->-r requirements.txt (line 7)) (1.12.1)
Requirement already satisfied: chardet<4,>=3.0.2 in /usr/local/lib/python3.8/dist-packages (from requests>=2.18.0->oslo.config>=5.2.0->oslo_concurrency->-r requirements.txt (line 7)) (3.0.4)
Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.8/dist-packages (from requests>=2.18.0->oslo.config>=5.2.0->oslo_concurrency->-r requirements.txt (line 7)) (2.10)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /usr/local/lib/python3.8/dist-packages (from requests>=2.18.0->oslo.config>=5.2.0->oslo_concurrency->-r requirements.txt (line 7)) (1.26.2)
Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.8/dist-packages (from requests>=2.18.0->oslo.config>=5.2.0->oslo_concurrency->-r requirements.txt (line 7)) (2020.11.8)
Using cached numpy-1.18.5-cp38-cp38-manylinux1_x86_64.whl (20.6 MB)
Using cached scikit_learn-1.0.2-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (26.7 MB)
Using cached matplotlib-3.4.2-cp38-cp38-manylinux1_x86_64.whl (10.3 MB)
Using cached pandas-1.2.4-cp38-cp38-manylinux1_x86_64.whl (9.7 MB)
Using cached scipy-1.7.3-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (39.3 MB)
Using cached statsmodels-0.13.5-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (9.9 MB)
Using cached appdirs-1.4.4-py2.py3-none-any.whl (9.6 kB)
Using cached oslo.concurrency-6.1.0-py3-none-any.whl (48 kB)
Using cached fasteners-0.19-py3-none-any.whl (18 kB)
Using cached oslo.config-9.6.0-py3-none-any.whl (132 kB)
Using cached oslo.i18n-6.4.0-py3-none-any.whl (46 kB)
Using cached oslo.utils-7.3.0-py3-none-any.whl (129 kB)
Using cached packaging-24.2-py3-none-any.whl (65 kB)
Using cached patsy-1.0.1-py2.py3-none-any.whl (232 kB)
Using cached pbr-6.1.1-py2.py3-none-any.whl (108 kB)
Using cached debtcollector-3.0.0-py3-none-any.whl (23 kB)
Using cached iso8601-2.1.0-py3-none-any.whl (7.5 kB)
Using cached netaddr-1.3.0-py3-none-any.whl (2.3 MB)
Using cached netifaces-0.11.0-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.whl (33 kB)
Using cached rfc3986-2.0.0-py2.py3-none-any.whl (31 kB)
Using cached stevedore-5.3.0-py3-none-any.whl (49 kB)
WARNING: Error parsing dependencies of pyzmq: Invalid version: 'cpython'
Installing collected packages: netifaces, diffdist, appdirs, rfc3986, pbr, packaging, numpy, netaddr, iso8601, fasteners, debtcollector, stevedore, scipy, patsy, pandas, oslo.i18n, matplotlib, statsmodels, scikit-learn, oslo.utils, oslo.config, oslo_concurrency
  Attempting uninstall: packaging
    Found existing installation: packaging 20.4
    Uninstalling packaging-20.4:
      Successfully uninstalled packaging-20.4
  Attempting uninstall: numpy
    Found existing installation: numpy 1.17.3
    Uninstalling numpy-1.17.3:
      Successfully uninstalled numpy-1.17.3
  Attempting uninstall: scipy
    Found existing installation: scipy 1.4.1
    Uninstalling scipy-1.4.1:
      Successfully uninstalled scipy-1.4.1
  Attempting uninstall: pandas
    Found existing installation: pandas 0.25.3
    Uninstalling pandas-0.25.3:
      Successfully uninstalled pandas-0.25.3
  Attempting uninstall: matplotlib
    Found existing installation: matplotlib 3.3.2
    Uninstalling matplotlib-3.3.2:
      Successfully uninstalled matplotlib-3.3.2
  Attempting uninstall: scikit-learn
    Found existing installation: scikit-learn 0.23.2
    Uninstalling scikit-learn-0.23.2:
      Successfully uninstalled scikit-learn-0.23.2
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
notebook 6.0.3 requires pyzmq>=17, which is not installed.
Successfully installed appdirs-1.4.4 debtcollector-3.0.0 diffdist-0.1 fasteners-0.19 iso8601-2.1.0 matplotlib-3.4.2 netaddr-1.3.0 netifaces-0.11.0 numpy-1.18.5 oslo.config-9.6.0 oslo.i18n-6.4.0 oslo.utils-7.3.0 oslo_concurrency-6.1.0 packaging-24.2 pandas-1.2.4 patsy-1.0.1 pbr-6.1.1 rfc3986-2.0.0 scikit-learn-1.0.2 scipy-1.7.3 statsmodels-0.13.5 stevedore-5.3.0
Computing plots for robust radius...
Files already downloaded and verified
Files already downloaded and verified
Encoder: models/mmcl/rbf/cnn_4layer_b_C_1_rbf_auto.pkl, eval_ckpt: models/linear_evaluate/linear_finetune_cnn_4layer_b_C_1_rbf_auto.pkl
/usr/local/lib/python3.8/dist-packages/torch/cuda/__init__.py:155: UserWarning: 
NVIDIA H100 with CUDA capability sm_90 is not compatible with the current PyTorch installation.
The current PyTorch install supports CUDA capabilities sm_37 sm_50 sm_60 sm_70 sm_75 sm_80 sm_86.
If you want to use the NVIDIA H100 GPU with PyTorch, please check the instructions at https://pytorch.org/get-started/locally/

  warnings.warn(incompatible_device_warn.format(device_name, capability, " ".join(arch_list), device_name))
Encoder: models/regular_cl/regular_cl_cnn_4layer_b_bs_32_lr_1e-3.pkl, eval_ckpt: models/linear_evaluate/linear_finetune_regular_cl_cnn_4layer_b_bs_32_lr_1e-3.pkl
Loading model with checkpoint: models/linear_evaluate/cifar10_cnn_4layer_b_adv2.pkl
Loading model with checkpoint: models/supervised/supervised_bs_32_lr_0.0001.pkl
Loaded verifiers
Iterating through the test dataset
Processing sigma: 0.1
Processing class name: cat
Traceback (most recent call last):
  File "compare_randomized_smoothing.py", line 171, in <module>
    update_results(
  File "compare_randomized_smoothing.py", line 121, in update_results
    rs_label, radius = verifier.certify(
  File "/workspace/randomized_smoothing/core.py", line 41, in certify
    counts_selection = self._sample_noise(x, n0, batch_size)
  File "/workspace/randomized_smoothing/core.py", line 95, in _sample_noise
    counts += self._count_arr(predictions.cpu().numpy(), self.num_classes)
  File "/workspace/randomized_smoothing/core.py", line 101, in _count_arr
    counts[idx] += 1
IndexError: index -4858292166798756609 is out of bounds for axis 0 with size 10
